### PR TITLE
[RDY] Changing creatures damage multipliers on hit

### PR DESCRIPTION
**Pull request by Firestorm01X2**
Current hit damage multilayer is:

Hit | Mult
-- | --
Grazing | 0-0.25
Standard | 0.5-1
Good | 1-1.5
Crit | 1.75-2.3
Head | 2.45-3.35

This PR proposed to change it to:

Hit | Mult
-- | --
Grazing | 0.5
Standard | 0.5-1
Good | 1-1.5
Crit | 1.5-2
Head | 2

Current graze hit damage multilayer with firearm is:
(0, 0.25)

Than mean that if you graze target than you get at best 25% of you weapon damage. But it is at best. Most of the time you get around 10% of your weapon damage.

I understand idea of graze hit by itself, but it current practical usage of it has no sense.

Even for sniper rifle grazing gives around 7 damage. In practice it is equivalent of wasted bullet. And it is for sniper riles. For smaller arms it looks like joke on player.
Also currently grazing hit may produce 0 damage that make no sense (you hit target after all).

This change give us:
* It proposes player to use snap shots instead of always aimed. Reduced headshot and critical hit damage should also make partial targeting more attractive;
* It give grazing hit meaning of hit. Cutting damage on half is already enough;
* It give player somewhat of minimal reliable damage on hit if player manage to hit at least somehow;
* Additionally this PR reduce critical hit and headshot damage. It may be useful to compensate increasing of grazing hit damage and normalize shot damage overall.

I understand that it sounds somewhat controversial, but anyway grazing damage multiplayer shouldn't be [0, 0.25]. That value is basically means missed shot.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Would be good to increase it more, but also decrease headshot and crit hit considerably.
Something like:

Hit | Mult
---|---
Grazing | 0.5
Standard | 0.5-1
Good | 1-1.5
Crit | 1.5-2
Head | 2

**Comment by Firestorm01X2** [comment]
Make sense.

**Comment by Firestorm01X2** [comment]
Done. Currently it is comparable with UFO 2 Terror from the depth. Is has damage roll 50%-150%

**Comment by kevingranade** [comment]
Tl;Dr, do not increase graze damage, it is working as intended. It is effectively a miss.

> Than mean that if you graze target than you get at best 25% of you weapon damage. But it is at best. Most of the time you get around 10% of your weapon damage.

Correct, and this is exactly how it is supposed to work.

> It proposes player to use snap shots instead of always aimed.

This is not an improvement, the system is designed to discourage snap shots. You should always spend time aiming unless you are extremely close to the target.

> Additionally this PR reduce critical hit and headshot damage. It may be useful to compensate increasing of grazing hit damage and normalize shot damage overall.

This part is fine.

**Comment by Coolthulhu** [comment]
> Correct, and this is exactly how it is supposed to work.

The "angle size" of grazing hit is twice the size of headshot. If it is basically a miss, it should be 0% of this size.

**Comment by kevingranade** [comment]
Feel free to reduce the angle at which grazes occur then.

Alternately load the thresholds and damage modifiers from json or something so it's moddable.

**Comment by Firestorm01X2** [comment]
I don't undertand why fo we neer graze if graze is practically miss? Why it shouldn't be miss then?
So it is basically hidden debuff to aiming.

**Comment by kevingranade** [comment]
No, it's a cosmetic buff to shooting, all the stats the game reports are
chance of regular hit or chance of good hit.

**Comment by Firestorm01X2** [comment]
So are tests based on idea that graze is miss?
